server/service/src/invoice/inbound_shipment/update/generate.rs

```diff
@@ -117,18 +128,16 @@ pub fn should_create_batches(invoice: &InvoiceRow, patch: &UpdateInboundShipment
         let new_invoice_status_index = new_invoice_status.index();
 
         new_invoice_status_index >= InvoiceRowStatus::Delivered.index()
-            && invoice_status_index < new_invoice_status_index
+            && invoice_status_index < InvoiceRowStatus::Delivered.index()
```

Comment:
Only return `true` if this inbound shipment hasn't been delivered yet.

Reply:
You can still add stock if status is `Delivered`....